extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/SpringCloudConfigServerClientConfigSourceProvider.java

```diff
@@ -28,23 +32,33 @@ public SpringCloudConfigServerClientConfigSourceProvider(SpringCloudConfigClient
         this.applicationName = applicationName;
         this.activeProfile = activeProfile;
 
-        springCloudConfigClientGateway = new DefaultSpringCloudConfigClientGateway(springCloudConfigClientConfig);
+        springCloudConfigClientGateway = new VertxSpringCloudConfigGateway(springCloudConfigClientConfig);
     }
 
     @Override
     public Iterable<ConfigSource> getConfigSources(ClassLoader forClassLoader) {
+        List<ConfigSource> result = new ArrayList<>();
+        List<Uni<?>> allUnis = new ArrayList<>();
```

Comment:
Looks like we only have one `Uni`, right?

If so there is no need to use a list nor to do a `combine` at the end of the method